Tools/autotest/arducopter.py

```diff
@@ -5636,6 +5668,8 @@ def fly_ship_takeoff(self):
         self.wait_groundspeed(0, 2)
 
     def test_parameter_validation(self):
+        # wait 10 seconds for initialisation
+        self.delay_sim_time(10)
```

Comment:
This is waiting for motors to be allocated, I guess?

Is there some state we can look for instead of adding a delay?

What happens *without* this?

Reply:
It's not the motor library but rather the GPS driver (when GPS_TYPE = Auto) which takes some time to figure out if there is a GPS present or not.  Without this delay the pre-arm fails because EKFSource library complains, "EKF3 requires GPS".  The problem is not that there is actually no GPS but rather that we're trying to arm before the GPS driver has figured out which GPS is being used.

This is a cut-and-paste of my comment in discord:

"I think what I actually need is an AP_DAL_GPS::get_num_enabled() method instead of num_sensors().  The problem is that the num_sensors() relies on a driver discovering an active GPS which takes some time and leads to a race condition (aka "flapping test").  What we really want in the EKFSources class is to check for a configuration inconsistency between GPS_TYPE and EK3_SRC1_POSXY/VELXY/VELZ.  I might ask Tridge tomorrow if he can help me add this somehow."

To which Tridge replied, "that makes a lot of sense"

We could wait to merge this until this change has been made to EKFSources but I don't think we should.  Then we could come back and remove this 10sec delay.